drivers/vmwarevsphere/cloudinit.go

```diff
@@ -0,0 +1,272 @@
+package vmwarevsphere
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/diskfs/go-diskfs/filesystem/iso9660"
+	"github.com/docker/machine/libmachine/log"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	isoName = "user-data.iso"
+	isoDir  = "cloudinit"
+)
+
+func (d Driver) cloudInit(vm *object.VirtualMachine) error {
+	if d.CloudInit != "" {
+		return d.cloudInitGuestInfo(vm)
+	}
+
+	if err := d.createCloudInitIso(); err != nil {
+		return err
+	}
+
+	dss, err := d.finder.DatastoreOrDefault(d.getCtx(), d.Datastore)
+	if err != nil {
+		return err
+	}
+
+	err = d.uploadCloudInitIso(vm, d.datacenter, dss)
+	if err != nil {
+		return err
+	}
+
+	err = d.mountCloudInitIso(vm, d.datacenter, dss)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d Driver) cloudInitGuestInfo(vm *object.VirtualMachine) error {
```

Comment:
This function eats a lot of errors, it would be nice to give something back to the user so they don't have to wait for the VM to come up before they find out there was a problem here.

Reply:
fixed with linter